contracts/beyondFramework/dca/base/DCABaseUpgradeable.sol

```diff
@@ -0,0 +1,1168 @@
+    }
+
+    function currentInvestQueueIndext() external view returns (uint8) {
+        return globalInvestQueue.current;
+    }
+
+    function getHistoricalGaugeAt(uint256 index)
+        external
+        view
+        returns (uint256, uint256)
+    {
+        return dcaHistory.gaugeByIndex(index);
+    }
+
+    function currentDcaHistoryIndex() external view returns (uint256) {
+        return dcaHistory.currentHistoricalIndex();
+    }
+
+    function isEmergencyExited() public view virtual returns (bool) {
+        return
+            bluechipInvestmentState == BluechipInvestmentState.EmergencyExited;
+    }
+
+    function depositToken() public view returns (IERC20Upgradeable) {
+        return depositTokenInfo.token;
+    }
+
+    // ----- Private Base Class Helper Functions -----
+    function _takeFee(uint256 amount) private returns (uint256 actualDeposit) {
+        // if fee is set to 0 then skip it
+        if (depositFee.fee == 0) {
+            return amount;
+        }
+
+        // actual deposit = amount * (100% - fee%)
+        actualDeposit = (amount * (100 - depositFee.fee)) / 100;
+
+        uint256 feeAmount = amount - actualDeposit;
+        if (feeAmount != 0) {
+            depositTokenInfo.token.safeTransfer(
+                depositFee.feeReceiver,
+                feeAmount
+            );
+        }
+    }
+
+    function _swapIntoBluechipAsset(uint256 amountIn)
+        private
+        returns (uint256)
+    {
+        return router.swapTokensForTokens(amountIn, depositToBluechipSwapPath);
+    }
+
+    function _swapIntoDepositAsset(uint256 amountIn) private returns (uint256) {
+        return router.swapTokensForTokens(amountIn, bluechipToDepositSwapPath);
+    }
+
+    function _getPassedInvestPeriods() private view returns (uint256) {
+        // solhint-disable-next-line not-rely-on-time
+        return (block.timestamp - lastInvestmentTimestamp) / investmentPeriod;
+    }
+
+    function _removeUserInvestmentFromHistory(
+        Position memory position,
+        uint256 passedInvestPeriods,
+        uint256 perPeriodInvestment
+    )
+        private
+        returns (uint256 bluechipInvestment, uint256 depositAssetInvestment)
+    {
+        // iterate over historical gauges since initial deposit
+        for (
+            uint256 j = position.investedAtHistoricalIndex;
+            j < position.investedAtHistoricalIndex + passedInvestPeriods;
+            j++
+        ) {
+            // total spent and received at selected investment day
+            (
+                uint256 totalAmountSpent,
+                uint256 totalAmountExchanged
+            ) = dcaHistory.gaugeByIndex(j);
+
+            // calculate user share for specified gauge
+            uint256 depositorBluechipShare = (perPeriodInvestment *
+                depositTokenScale) / totalAmountSpent;
+            // calculate amount that user ownes in current gauge
+            uint256 depositorOwnedBluechip = (totalAmountExchanged /
```

Comment:
can we just do?
depositorOwnedBluechip = totalAmountExchanged*perPeriodInvestment/totalAmountSpent?

Reply:
perPeriodInvestment and totalAmountSpent have the same decimal count, so they would just cancel each other, and the final decimal count would be that of totalAmountExchanged